### PR TITLE
Unit Test

### PR DESCRIPTION
**Pull request by neetp7k9**
The unit test is based on the Unity C test Framework
The test includes the IO test and the Driver test
Please fill in the configuration in sx_wifi_test_conf.h and sx_wifi_test_key.h before running the test

### CODE REVIEW

File: Src/main.c [hunk 1]
```diff
@@ -59,10 +59,10 @@ DMA_HandleTypeDef hdma_usart3_rx;
 /* Private function prototypes -----------------------------------------------*/
 void SystemClock_Config(void);
 static void MX_GPIO_Init(void);
-static void MX_DMA_Init(void);
+void MX_DMA_Init(void);
 static void MX_USART1_UART_Init(void);
```
Comment (bitwise-panda) [inline Src/main.c]: MX_USART1_UART_Init is a static void why is this different from MX_USART2_UART_Init, MX_USART3_UART_Init?

What is the visibility you are trying to achieve for each function?  What is calling it?

File: sx_59hls/include/sx_wifi.h [hunk 2]
```diff
@@ -13,6 +21,13 @@
 #include "sx_wifi_conf.h"
 
 /* Exported Constants --------------------------------------------------------*/
+
+#define SX_WIFI_MAX_SSID_NAME_SIZE                  32            /**< SSID length */
+#define SX_WIFI_MAX_PSWD_NAME_SIZE                  32            /**< Password length */
```
Comment (bitwise-panda) [inline sx_59hls/include/sx_wifi.h]: Consider changing the name SX_WIFI_MAX_PSWD_NAME_SIZE = SX_WIFI_MAX_PASSWORD_NAME_SIZE so it's more readable / maintainable.

File: sx_59hls/include/sx_wifi.h [hunk 3]
```diff
@@ -13,6 +21,13 @@
...
+#define SX_WIFI_PRODUCT_NAME_SIZE                   32            /**< Product name length */
+#define SX_WIFI_FW_REV_SIZE                         24            /**< Firmware version length */
+#define SX_WIFI_API_REV_SIZE                        16            /**< API version length */
+
 #define SX_WIFI_PAYLOAD_SIZE     1200
 #define SX_WIFI_GUARD_TIME_UINT     5
```
Comment (bitwise-panda) [inline sx_59hls/include/sx_wifi.h]: Documentation about what SX_WIFI_GUARD_TIME_UINT would be useful.

File: Src/main.c (hunk 1 above)
Comment (neetp7k9) [reply Src/main.c]: MX_USART1_UART_Init  -> Log printing UART PORT
 MX_USART2_UART_Init, MX_USART3_UART_Init -> SX-59HLS WiFi Related UART, CMD and Data Port

**Pull request merged**